core/modules/util/ThreadPool.cc

```diff
@@ -137,17 +145,25 @@ PoolEventThread::Ptr CommandThreadPool::getAndNullPoolEventThread() {
 }
 
 
+ThreadPool::Ptr ThreadPool::newThreadPool(unsigned int thrdCount, unsigned int maxThreadCount,
+                                          CommandQueue::Ptr const& q, EventThreadJoiner::Ptr const& joiner) {
+    Ptr thp(new ThreadPool(thrdCount, maxThreadCount, q, joiner)); // private constructor
+    thp->_resize();
+    return thp;
+}
+
+
 ThreadPool::Ptr ThreadPool::newThreadPool(unsigned int thrdCount,
-      CommandQueue::Ptr const& q, EventThreadJoiner::Ptr const& joiner) {
-    Ptr thp(new ThreadPool(thrdCount, q, joiner)); // private constructor
+                                          CommandQueue::Ptr const& q, EventThreadJoiner::Ptr const& joiner) {
+    Ptr thp(new ThreadPool(thrdCount, thrdCount+1, q, joiner)); // private constructor
```

Comment:
A choice of having the second parameter incremented by 1 could be explained in a comment. It's not clear to a reader why it's +1 and not +2, etc.